components/lcm/org.wso2.carbon.apimgt.lifecycle.manager.core/src/main/java/org/wso2/carbon/apimgt/lifecycle/manager/core/internal/ServiceReferenceHolder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.lifecycle.manager.core.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.utils.ConfigurationContextService;
+
+/**
+ * This class provides single instance in order to hold references to other services used by
+ * lifecycle service component.
+ */
+public class ServiceReferenceHolder {
+    private static Log log = LogFactory.getLog(ServiceReferenceHolder.class);
+    private static ServiceReferenceHolder instance;
+    private  ConfigurationContextService contextService;
+
+    private ServiceReferenceHolder() {
+
+    }
+
+    public static ServiceReferenceHolder getInstance() {
```

Comment:
Instead of this, to impose singleton method u can initialize the ServiceReferenceHolder object at the place it is declared and return that instance in this class like below.  Its much cleaner. 

private static ServiceReferenceHolder serviceRefHolderInstance = new ServiceReferenceHolder();

public static ServiceReferenceHolder getInstance() {
  return serviceRefHolderInstance;
}